### ISSUE TITLE
Conversion from pytket to qiskit fails for circuits containing conditions on a partial register

### ISSUE THREAD

**Issue by cqc-alec** [CQCL/pytket-qiskit#43]
```python
from pytket.circuit import Circuit
from pytket.extensions.qiskit import tk_to_qiskit

c = Circuit(1, 2)
c.X(0, condition_bits=[0], condition_value=1)

tk_to_qiskit(c)
```

Output:

```
NotImplementedError: OpenQASM conditions must be an entire register
```

The error message is misleading because we are not using OpenQASM for the conversion. It should be possible to convert such circuits to qiskit.

**Comment by CalMacCQ** [comment]
Interestingly there are several error messages that are raised related to non qasm-compilant circuits. See [here](https://github.com/CQCL/pytket-qiskit/blob/develop/pytket/extensions/qiskit/qiskit_convert.py#L652) and the corresponding tests [here](https://github.com/CQCL/pytket-qiskit/blob/develop/tests/qiskit_convert_test.py#L403). This is indeed strange as we aren't using QASM for the conversion.

If I remove these lines of code it seems like qiskit circuits are converted incorrectly.

```python
from pytket.circuit import Circuit
from pytket.extensions.qiskit import tk_to_qiskit, qiskit_to_tk

tkc = Circuit(1, 2)
tkc.X(0, condition_bits=[0], condition_value=1)

qc = tk_to_qiskit(tkc)

tkc2 = qiskit_to_tk(qc)
```
`tkc` looks like this
<img width="210" alt="Screenshot 2023-08-10 at 11 15 46" src="https://github.com/CQCL/pytket-qiskit/assets/93673602/f6782f54-899e-4199-857d-d817fd62fdf6">

`tkc2` looks like this... (the X is conditoned on both of the bits).
<img width="214" alt="Screenshot 2023-08-10 at 11 15 58" src="https://github.com/CQCL/pytket-qiskit/assets/93673602/de13d5bc-e53b-42b6-badf-39bc9bca3440">

**Comment by cqc-alec** [comment]
Closed by #158 .

**Comment by cqc-alec** [comment]
Issue of inability to convert to qiskit remains.
